src/main/java/com/redhat/devtools/lsp4ij/features/AbstractLSPInlayHintsProvider.java

```diff
@@ -65,47 +66,30 @@ public final InlayHintsCollector getCollectorFor(@NotNull PsiFile psiFile,
         final long modificationStamp = psiFile.getModificationStamp();
         return new FactoryInlayHintsCollector(editor) {
 
-            private boolean processed;
-
             @Override
             public boolean collect(@NotNull PsiElement psiElement, @NotNull Editor editor, @NotNull InlayHintsSink inlayHintsSink) {
-                if (processed) {
-                    // Before IJ 2023-3, FactoryInlayHintsCollector#collect(PsiElement element.. is called once time with PsiFile as element.
-                    // Since IJ 2023-3, FactoryInlayHintsCollector#collect(PsiElement element.. is called several times for each token of the PsiFile
-                    // which causes the problem of codelens/inlay hint which are not displayed because there are too many call of LSP request codelens/inlayHint which are cancelled.
-                    // With IJ 2023-3 we need to collect LSP CodeLens/InlayHint just for the first call.
-                    return false;
+                if (!(psiElement instanceof PsiFile psiFile)) {
+                    return true;
```

Comment:
Return true fixes the duplicate problem? Could you explain more how you have fixed duplicate inlay hints please.

Reply:
The `collect` method is called once per file, so the `processed` check can be replaced with a check for whether the `psiElement` is a `PsiFile`, and that will prevent the duplicate inlay hints bug.
Return true is there for consistency with other examples from open source intellij code (Apache 2.0), which always returns true, see: https://github.com/JetBrains/intellij-community/blob/9d33ede5a92b3d533fe7c1a906d6a425067902b0/plugins/markdown/core/src/org/intellij/plugins/markdown/editor/tables/ui/MarkdownTableInlayProvider.kt#L30